napari/_qt/qt_main_window.py

```diff
@@ -635,6 +635,68 @@ def remove_dock_widget(self, widget: QWidget):
         # rather than directly using _add_viewer_dock_widget
         _dw.deleteLater()
 
+    def add_function_widget(
+        self,
+        function,
+        *,
+        magic_kwargs=None,
+        name: str = '',
+        area: str = 'bottom',
+        allowed_areas=None,
+        shortcut=None,
+    ):
+        """Turn a function into a dock widget via magicgui.
+
+        Parameters
+        ----------
+        function : callable
+            Function that you want to add.
+        magic_kwargs : dict, optional
+            Keyword arguments to the magicgui function that
```

Comment:
```suggestion
            Keyword arguments to :func:`magicgui.magicgui` that
```

(this would provide a direct link if we add magicgui intersphinx to our docs)